### PR TITLE
Fix development mode for the 'main-react' branch

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
Fix development mode for the main-react branch.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18824

### CODE REVIEW

File: devfile.yaml [hunk 1]
```diff
@@ -1,40 +1,49 @@
 apiVersion: 1.0.0
 metadata:
-  name: dashboard-next
+  generateName: che-dashboard-react
```
Comment (akurinnoy) [inline devfile.yaml:3]:
```suggestion
  generateName: che-dashboard-react-
```

File: devfile.yaml [hunk 2]
```diff
@@ -1,40 +1,49 @@
...
+        port: 3000
+    command:
+      - tail
+      - '-f'
+      - /dev/null
+    memoryLimit: 2Gi
     type: dockerimage
-    alias: ud-dev
-    image: 'okurinnyi/che-dashboard-dev:nightly'
+    alias: dash-dev
```
Comment (akurinnoy) [inline devfile.yaml:21]: Please consider either reverting alias name or choosing another, less confusing alias, e.g. `dashboard-dev`.

File: devfile.yaml [hunk 3]
```diff
@@ -1,40 +1,49 @@
...
+    image: 'docker.io/node:10.22.1'
```
Comment (akurinnoy) [inline devfile.yaml:22]:
```suggestion
    image: 'docker.io/node:12'
```

File: devfile.yaml (hunk 1 above)
Comment (olexii4) [reply devfile.yaml]: I have fixed

File: devfile.yaml (hunk 2 above)
Comment (olexii4) [reply devfile.yaml]: OK. I have fixed it.

File: devfile.yaml (hunk 3 above)
Comment (olexii4) [reply devfile.yaml]: OK. I have fixed it.

Comment (olexii4) [reply devfile.yaml]: https://github.com/eclipse/che-dashboard/blob/main-react/package.json#L21

Comment (olexii4) [reply devfile.yaml]:
```
"node": ">=10.20.1"
```

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**